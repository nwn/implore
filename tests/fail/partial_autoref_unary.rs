use implore::impl_op;

#[derive(Copy, Clone, PartialEq, Eq, Debug)]
struct X;

#[impl_op(autoref)]
fn neg(rhs: X) -> X {
    rhs
}

fn partial_negation() {
    assert_eq!(X, -X);
    assert_eq!(X, -&X);
}

fn main() {}
