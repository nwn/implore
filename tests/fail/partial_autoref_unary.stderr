warning: option `autoref` has no effect
 --> $DIR/partial_autoref_unary.rs:6:11
  |
6 | #[impl_op(autoref)]
  |           ^^^^^^^

error[E0600]: cannot apply unary operator `-` to type `&X`
  --> $DIR/partial_autoref_unary.rs:13:19
   |
13 |     assert_eq!(X, -&X);
   |                   ^^^ cannot apply unary operator `-`
   |
   = note: an implementation of `std::ops::Neg` might be missing for `&X`
